telemetry-aware-scheduling/deploy/cluster-api/docker/capi-docker.md

```diff
@@ -0,0 +1,160 @@
+- Change the `dnsPolicy` of the scheduler to `ClusterFirstWithHostNet`
+- Place `KubeSchedulerConfiguration` into control plane nodes and pass the relative CLI flag to the scheduler.
+- Change the behavior of the pre-existing patch application of `/spec/template/spec/kubeadmConfigSpec/files` in `ClusterClass` 
+such that our new patch is not ignored/overwritten. For some more clarification on this, see [this issue](https://github.com/kubernetes-sigs/cluster-api/pull/7630).
+
+You will also need to add a label to the `Cluster` resource of your new cluster to allow ClusterResourceSets to target
+it (see `cluster-patch.yaml`). Simply add a label `scheduler: tas` in your `Cluster` resource present in `your-manifests.yaml`.
+
+3. You will need to prepare the Helm Charts of the various components and join the TAS manifests together for convenience:
+
+First, under `telemetry-aware-scheduling/deploy/charts` tweak the charts if you need (e.g.
+additional metric scraping configurations), then render the charts:
+
+  ```bash
+  helm template ../charts/prometheus_node_exporter_helm_chart/ > prometheus-node-exporter.yaml
+  helm template ../charts/prometheus_helm_chart/ > prometheus.yaml
+  helm template ../charts/prometheus_custom_metrics_helm_chart > prometheus-custom-metrics.yaml
+  ```
+
+You need to add namespaces resources, else resource application will fail. Prepend the following to `prometheus.yaml`:
+
+  ```bash
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: monitoring
+  labels:
+    name: monitoring
+  ````
+
+Prepend the following to `prometheus-custom-metrics.yaml`:
+  ```bash
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: custom-metrics
+  labels:
+    name: custom-metrics
+  ```
+
+The custom metrics adapter and the TAS deployment require TLS to be configured with a certificate and key.
+Information on how to generate correctly signed certs in kubernetes can be found [here](https://github.com/kubernetes-sigs/apiserver-builder-alpha/blob/master/docs/concepts/auth.md).
+Files ``serving-ca.crt`` and ``serving-ca.key`` should be in the current working directory.
+
+Run the following:
+
+  ```bash
+  kubectl -n custom-metrics create secret tls cm-adapter-serving-certs --cert=serving-ca.crt --key=serving-ca.key -oyaml --dry-run=client > custom-metrics-tls-secret.yaml
+  kubectl -n default create secret tls extender-secret --cert=serving-ca.crt --key=serving-ca.key -oyaml --dry-run=client > tas-tls-secret.yaml
+  ```
+
+**Attention: Don't commit the TLS certificate and private key to any Git repo as it is considered bad security practice! Make sure to wipe them off your workstation after applying the relative Secrets to your cluster.**
+
+You also need the TAS manifests (Deployment, Policy CRD and RBAC accounts) and the extender's "configmapgetter"
+ClusterRole. We will join the TAS manifests together, so we can have a single ConfigMap for convenience:
+
+  ```bash
+  yq '.' ../tas-*.yaml > tas.yaml
```

Comment:
I don't recall exactly why this one was left out (probably it was never spotted because GCP and Docker clusters worked fine without it). I guess it's better to include it since it's part of the Helm chart.